app/modifiers/is-dark-mode.ts

```diff
@@ -1,56 +1,139 @@
-// Invokes a callback when the dark mode setting changes (and on initial load).
-// Usage: <div {{is-dark-mode this.handleDarkModeUpdated}}></div>
 import { registerDestructor } from '@ember/destroyable';
-import { action } from '@ember/object';
-import { task } from 'ember-concurrency';
+import { service } from '@ember/service';
+import { task, timeout } from 'ember-concurrency';
 import Modifier from 'ember-modifier';
+import type DarkModeService from 'codecrafters-frontend/services/dark-mode';
 
+/**
+ * Enable or disable the DOM Mutation Observer
+ */
+const DOM_MUTATION_OBSERVER_ENABLED: boolean = false; // TODO: Figure out an efficient way to enable the observer while avoiding too many calls
+
+/**
+ * Enable or disable subscribing to `DarkModeService.isEnabled` changes
+ */
+const DARK_MODE_UPDATE_LISTENER_ENABLED: boolean = true;
+
+/**
+ * Method to use for delaying `triggerCallback` task execution:
+ * - `restartable` cancels all queued calls before they have a chance to execute, executes only the latest call: causes much less unnecessary task executions
+ * - `keepLatest` always executes first & last queued calls, cancelling the ones in-between: might be more reliable but always triggers rendering twice
+ */
+const CALLBACK_TASK_TIMEOUT_METHOD: 'keepLatest' | 'restartable' = 'restartable';
```

Comment:
@VasylMarchuk same as above - let's remove any configuration options like this and just choose one

Reply:
Removed this helper completely